docker/bump_version.sh

```diff
@@ -19,27 +17,12 @@ else
   PART="${1}"
 fi
 
-CURRENT_BRANCH=$(git branch --show-current)
-CURRENT_STATUS=$(git status --short --untracked-files=no)
-
-if [[ "${CURRENT_BRANCH}" != "${DEFAULT_BRANCH}" ]]; then
-  echo "A version bump must be run from the default branch."
-  echo "Run 'git switch ${DEFAULT_BRANCH}'"
-  exit 2
-elif [[ "$CURRENT_STATUS" != "" ]]; then
-  echo "The working tree has uncommitted changes."
-  echo "Commit or stash the changes before running a version bump."
-  exit 3
-fi
-# Ensure on default branch
-git switch --no-guess main
+
 # Capture value of new version
-NEW_VERSION=$(bump2version --dry-run --list "${PART}" | grep new_version | sed -r s,"^.*=",,)
-# Create bump branch
-BUMP_BRANCH_NAME="bump_version_to_${NEW_VERSION}"
-git switch -c "${BUMP_BRANCH_NAME}"
+NEW_VERSION=$(bump2version --dry-run --allow-dirty --list "${PART}" | grep new_version | sed -r s,"^.*=",,)
+
 # Update files
-bump2version "${PART}"
+bump2version "${PART}" --allow-dirty
```

Comment:
I recommend not adding `--allow-dirty` as this could result in accidentally committing files. If we leave it out, the assumption would be that all version bumps will come as independent PRs with only the changes made by the bumpversion script (and any subsequent changes).

Is there a compelling reason to add it?

Reply:
I removed the parts of the bump version script that actually do committing and branch checking, so the script only updates the version and leaves all of the git tasks to the PR author. 

This makes it possible to push a change that includes a version bump - yes, you can do this if you run the bump version script first, but that is too easy to forget to do (and author may not know they'd like the version until after they've already added a few commits to the branch). I think this is safe, since PR authors will see and need to address merge conflicts before merging.

Reply:
Sorry for the late reply - based on the [bumpversion config](https://github.com/wayfair-incubator/pygitops/blob/56cdfb6b6896c61cd5c759b46726893e467d0ea0/.bumpversion.cfg#L3), bump2version will still commit the files it changes ([bump2version docs](https://github.com/c4urself/bump2version#commit--true--false) on the subject).

The [code you removed](https://github.com/wayfair-incubator/pygitops/pull/203/files#diff-822a8f45e842ea1317f77c4c1c49ba486d6eb7fc858332ebb5a8722012cb9698L53) just changed the changelog and added that to the previous commit (the one that bumpversion would have just made).

So, as currently implemented, bumping a version using this script will:

1. Create a commit bumping the versions in [these files](https://github.com/wayfair-incubator/pygitops/blob/56cdfb6b6896c61cd5c759b46726893e467d0ea0/.bumpversion.cfg#L6-L12)
2. Change the changelog and leave it uncommitted (presumably for review from the author)

Sorry - I should have asked a clarifying question about this flow, but I thought this was intended. It sounds to me like we should either:

- Remove `--allow-dirty` and re-add the code [to add changelog to bumpversion commit](https://github.com/wayfair-incubator/pygitops/pull/203/files#diff-822a8f45e842ea1317f77c4c1c49ba486d6eb7fc858332ebb5a8722012cb9698L51)

Or:

- Not [commit](https://github.com/wayfair-incubator/pygitops/blob/56cdfb6b6896c61cd5c759b46726893e467d0ea0/.bumpversion.cfg#L3) bumpversion changes